base/reduce.jl

```diff
@@ -306,25 +306,29 @@ with reduction `op` over an empty array with element type of `T`.
 
 If not defined, this will throw an `ArgumentError`.
 """
-reduce_empty(op, T) = _empty_reduce_error()
-reduce_empty(::typeof(+), T) = zero(T)
+reduce_empty(op, ::Type{T}) where T = _empty_reduce_error()
```

Comment:
AFAICT this is not a signature-based invalidation, this is one that came from `jl_method_table_disable` which seems to have `Base.delete_method` as its only caller. I don't have a very good handle on what is going on here, sorry.

One interesting thing is that the inferred return type is `Union{}` (even before loading FixedPointNumbers):
```julia
julia> code_typed(Base.reduce_empty, (Base.BottomRF{typeof(max)}, Type{VersionNumber}); optimize=false)
1-element Array{Any,1}:
 CodeInfo(
1 ─ %1 = Base.getproperty(op, :rf)::Core.Compiler.Const(max, false)
│        Base.reduce_empty(%1, T)::Union{}
└──      Core.Compiler.Const(:(return %2), false)::Union{}
) => Union{}
```